src/HeaderData.php

```diff
@@ -180,4 +187,81 @@ public function returnVaryHeader($key): array {
     return ['vary' => $vary_header_array];
   }
 
+  /**
+   * Get the singleton instance for the global functions API.
+   */
+  private static function getInstance(): HeaderData {
+    if (self::$headerData == null) {
+      self::$headerData = new HeaderData();
+    }
+ 
+    return self::$headerData;
+  }
+
+  /**
+   * Sets the given header data associated with the global functions.
+   * 
+   * @param $data array (optional)
+   *   The array data to set when reading header data. Defaults to $_SERVER.
+   */
+  public static function setHeaderData(array $data = null): void {
+    self::$headerData = new HeaderData($data);
+  }
+
+  /**
+   * Gets the global header data based on the given key.
+   *
+   * @param string $key
+   *   Key for the header.
+   *
+   * @return string
+   *   Returns header value.
+   *
+   * @see getHeader()
+   */
+  public static function header($key) {
+    return self::getInstance()->getHeader($key);
+  }
+
+  /**
+   * Parses a global header by key using a specified regex.
+   *
+   * @param string $key
+   *   Key for the header.
+   *
+   * @return array|string
+   *   Returns important parts of header string.
+   *
+   * @see parseHeader()
+   */
+  public static function parse($key) {
+    return self::getInstance()->parseHeader($key);
+  }
+
+  /**
+   * Gets the global personalizaition object.
+   *
+   * @return array
+   *   Returns object with data used for personalization.
+   * 
+   * @see returnPersonalizedObject()
+   */
+  public static function personalizationObject() {
+    return self::getInstance()->returnPersonalizationObject();
```

Comment:
@jazzsequence Unfortunately PHP won't let us name the global functions the same as the instance functions. This means that $headerData->returnVaryHeader($key) has become HeaderData::varyHeader($key). Can you think of a better convention around these, @jazzsequence?